LB editor theming updates and header lock.

cory-skeers opened the pull request:
Resolves #2029, resolves #1523, resolves #1908, resolves #2046, resolves https://github.com/uiowa/uids/issues/274

Work toward providing a closer match between the LB 'preview' available in the editor and the final produced page.

### Testing
- `composer install`
- Sync a site.
- Check that the issues described in #2029 are resolved.
- Add a local webmaster and masquerade as them. Check that the header section is locked while masquerading but not as an admin.

cory-skeers committed:
Remove section padding for edge-to-edge styling. Remove margin between sections.

cory-skeers committed:
Remove top/bottom margins between sections.

cory-skeers committed:
Remove extra padding from add_block.

cory-skeers committed:
More padding removal.

cory-skeers committed:
Change add-block and -section outlines to &:hover, and match coloring.

cory-skeers committed:
Add outline/dropshadow to section buttons to match add-block buttons.

cory-skeers committed:
Re-added slight margin to sections. A little separation seems clearer.

cory-skeers committed:
Increase size of add-section button text to match add-block.

cory-skeers committed:
Removed block outline hover and added hover to section (make it clear which section the block is added to. Darkened non-hover section outline and used uids color variable.

cory-skeers committed:
Re-adjusted section outlines/margins/padding.

cory-skeers committed:
Merge branch 'main' into lb-editor-theming

Commit:
Require drupal/layout_builder_lock * 1.0.0-rc4

Commit:
Enable layout_builder_lock and configure

cory-skeers committed:
Merge branch 'main' into lb-editor-theming

Commit:
Export layout lock settings to collegiate

cory-skeers committed:
Added some space back to the add-section block. Moved the section outlines farther out.

Commit:
Merge branch 'lock-header' into lb-editor-theming

Commit:
Reduce admin hero height to 30vh at all viewport widths

richardbporter commented on docroot/themes/custom/uids_base/scss/admin.scss:
```diff
@@ -66,17 +73,22 @@
+  }
+
+  .hero {
+    height: 30vh;
```
I'm not sure if this is the best way to address part of #1908.

bspeare replied:
I think we want to do something like `height: auto;` for now until the Hero refactor is complete.  The `vh` has added specifically for the home page video to stay within the viewport.

richardbporter replied:
Yes that works much better. 👍

Commit:
Patch lb_direct_add

https://www.drupal.org/files/issues/2020-08-14/add-button-if-exists-3165272-2.patch

Commit:
Update lock file

bspeare committed:
Adjusted Hero size

Commit:
Set LB hero to height auto

richardbporter committed:
Merge branch 'main' into lb-editor-theming

bspeare committed:
Merge branch 'lb-editor-theming' of https://github.com/uiowa/uiowa into lb-editor-theming

bspeare committed:
Merge branch 'lb-editor-theming' of https://github.com/uiowa/uiowa into lb-editor-theming

bspeare committed:
Testing this commit

bspeare committed:
Removed add block sections from layout onecol background template

cory-skeers committed:
Merge branch 'main' into lb-editor-theming

richardbporter commented on docroot/themes/custom/uids_base/templates/layouts/onecol--background/layout--onecol--background.html.twig:
```diff
@@ -21,16 +21,15 @@
   } %}
     {% block hero_image %}
-      {{ content.background }}
+      {{ content.background | without('layout_builder_add_block')}}
```
I think these can be removed now. The buttons should not show up with the patch applied to lb_direct_add.

bspeare committed:
Removed without statement from template

Commit:
Create has_node_image layout proprocess var for onecol background template

richardbporter committed:
Merge branch 'main' into lb-editor-theming

Commit:
Merge branch 'main' into lb-editor-theming

bspeare committed:
Adjusted add section and edit block buttons

bspeare committed:
Adjusted z-index

bspeare committed:
Adjusted background color of Add Section

bspeare committed:
Further adjustments to Add Section button

bspeare committed:
Adjusted Configure Section button to look more like Add Section button

bspeare committed:
Added top margin to drop button and negative margins to add section block so that it aligns with dropbutton text

bspeare committed:
Adjusted dropbutton alignment for arrow

bspeare committed:
Removed thick outline from Configure Block button

bspeare committed:
Adjusted color of button link within dropbutton and Add Section button so that the colors are separate from the layout builder content

bspeare committed:
Update z-index for Configure Section button

bspeare committed:
Updated styles with Sean's refactor

bspeare committed:
Additional styles cleanup

bspeare committed:
Updated Configure Section button styles

bspeare committed:
Adjusted contextual trigger drop shadow

pyrello committed:
Tightening up vertical space. Removing redundant outlines/borders.

pyrello committed:
Adjust hidden block message styling. Fix button styling for 'Add section'.

Commit:
Style off-canvas autocomplete dropdown options so they are readable, resolves #2007

pyrello committed:
Added some white space around the 'Add section' button.

pyrello committed:
Merge branch 'lb-editor-theming' of github.com:uiowa/uiowa into lb-editor-theming

Commit:
Remove unnecessary font family style from autocomplete anchors

bspeare committed:
Updated global.scss padding options to target lb__container

pyrello committed:
Added some initial block interaction indication styles.

pyrello committed:
Merge branch 'lb-editor-theming' of github.com:uiowa/uiowa into lb-editor-theming

Commit:
Fix brain background pattern in LB editor preview

pyrello committed:
SCSS clean up. Correcting syntax for template file.

pyrello committed:
Fixing merge conflicts.

bspeare committed:
Added background image patterns to admin.scss until a better solution for the white background on layout-builder-block can be fixed

richardbporter committed:
Merge branch 'main' into lb-editor-theming

pyrello committed:
Prevent attributes from passing through to hero heading.

pyrello committed:
Added padding for title, body, and links placeholder blocks.

pyrello committed:
Prevented CTA from passing attributes through to headline.

richardbporter reviewed (commented):
Does a15842d resolve https://github.com/uiowa/uiowa/issues/2046?

pyrello committed:
Fixing title attribute issue on event block type, refactoring SCSS.

pyrello committed:
Moved to dev version of lb_direct_add and removed the patch.

pyrello committed:
Added padding and color indication for all 'placeholder' areas. Added outlines around blocks that show and are more defined when hovering over a section.

pyrello committed:
Removing background color and horizontal padding from placeholder blocks.

richardbporter committed:
Merge branch 'main' into lb-editor-theming

pyrello committed:
Added event subscriber to allow for adding a placeholder class to layout builder blocks using placeholder text.

GaryRidgway committed:
Merge branch 'main' into lb-editor-theming

pyrello committed:
Removed unnecessary new line. Finalizing treatment of blocks.

pyrello committed:
Fixing phpcs errors.

bspeare committed:
Removed css forcing contextual links to appear visible by default

richardbporter committed:
Merge branch 'main' into lb-editor-theming

richardbporter reviewed (approved):
Nice! I thought of that but forgot to mention it here.
